internal/k8s/service/resolver.go

```diff
@@ -321,9 +320,83 @@ func (r *backendResolver) findGlobalCatalogService(service *corev1.Service) (*Re
 	return nil, nil
 }
 
-// this will resolve a service based on our future CRD
-func (r *backendResolver) findConsulMeshService(name types.NamespacedName) {
-	mapped := r.mapper(name.Namespace)
+func (r *backendResolver) consulServiceForMeshService(ctx context.Context, namespacedName types.NamespacedName) (*ResolvedReference, error) {
+	var err error
+	var resolved *ResolvedReference
```

Comment:
Could use named return values here, but I haven't noticed whether that's a thing we do/don't do in this repo

Reply:
I'm not really a fan of named returns unless you're using at least 3 return values, but maybe that's just me, I remember their being some org-wide style guidelines that we may want to follow here, but I'm assuming either way there's not totally a hard or fast rule on it.

Reply:
It's not something I feel strongly about, just a thought I had as I read through